### ISSUE TITLE
Crosswalk + Native Page Transitions doesn't work

### ISSUE THREAD

**Issue by duke111** [Telerik-Verified-Plugins/NativePageTransitions#55]
I tried the Native Page Transitions  plugin together with the Cordova Crosswalk Webview, but only the flip transition seems to work. For slide and fade I have to click twice to get the transition. But then also the flip start flickering. Is there any solution?

**Comment by EddyVerbruggen** [comment]
Can you provide a few more details as I very recently fixed these issues.

A test project would help, along with the spec of the devices you tested this with.

Thanks!

**Comment by duke111** [comment]
Hi Eddy, first of all my compliments for your great plugin.
I tried the same project on Cordova (3.8)+ ios and it's ok.The same result with Cordova (4.02) + android: everything smooth.
But if I add the cordova-plugin-crosswalk-webview it's unusable.
I used a Galaxy S2 and a Sony tablet s for my tests. (android 4.1.2 and 4.0.3, respectively).
Apparently the  problem is the mix of the crosswalk plugin + the Native page transitions plugin.
Please let me know if you need only the html test project o something else.
Thanks in advance!




----Messaggio originale----

Da: notifications@github.com

Data: 12/06/2015 12.12

A: "Telerik-Verified-Plugins/NativePageTransitions"<NativePageTransitions@noreply.github.com>

Cc: "duke111"<fabio.fracassini@libero.it>

Ogg: Re: [NativePageTransitions] Crosswalk + Native Page Transitions doesn't work (#55)




Can you provide a few more details as I very recently fixed these issues.


A test project would help, along with the spec of the devices you tested this with.


Thanks!

**Comment by EddyVerbruggen** [comment]
Hmm, I can test with an S2 next monday. You are using the latest version (0.4.1), right?

That html will certaintly help, thx!

**Comment by duke111** [comment]
Yes 0.4.1 and all the plugins and cordova are updated, I rechecked to be sure.
As I told you before, the flip transition is ok. The problem is with slide and fade that break everything, but only in crosswalk (not in the normal  webview).  

However this is the link to the test project I used:
https://www.dropbox.com/s/c0qggdpdytu03df/TEST%20PROJECT.zip?dl=0

Can you please confirm me that your plugin works ok in your tests with the cordova-plugin-crosswalk-webview, because if flip works, it's hard for me to think that is a problem of device (2 of them).

Many thanks for your support

**Comment by EddyVerbruggen** [comment]
@duke111 Will take a look, thx. One last check: do you have hardware acceleration enabled in your AndroidManfest.xml? Does it make any difference with or without?

**Comment by duke111** [comment]
Yes I had it enabled by default. Just tried to disable it: no change.
And before I tried to set xwalk_animatable false/true in the preferences of the crosswalk plugin and no change again.

To be sure it wasn't a my mistake,  I also tried to change in my javascript only the flip/slide/fade in the same part of code (changing only the extension '.flip' etc.): flip ok, fade/slide broken.

I think there is something different in how work flip and fade/slide.

**Comment by EddyVerbruggen** [comment]
It's probably a problem in lower Android versions, I will look at it soon!

**Comment by bau720123** [comment]
+1
I have the same problem
I am using PGB(build.phonegap.com) 5.1.1
when I am not using crosswalk，all work perfect
but if I am using the crosswalk，all fail
https://build.phonegap.com/plugins/3347

**Comment by Menardi** [comment]
This may not be the issue @duke111 was having, but v1.3.0 (released August 28, 2015) of cordova-plugin-crosswalk-webview set ANIMATABLE_XWALK_VIEW to be false by default. This was causing the transitions not to work for me. To make the Crosswalk WebView animatable, add this line to your config.xml:

    <preference name="CrosswalkAnimatable" value="true" />

**Comment by bau720123** [comment]
hi @Menardi 
thanks for your reply
so...what cordova version are you using ?
us PGB (build.phonegap.com) support this attribute (CrosswalkAnimatable) ?

**Comment by Menardi** [comment]
Yes, when I add that to config.xml the transitions work perfectly. I don't know about Phonegap Build - if it lets you edit your config.xml file then you should have no problem.

**Comment by EddyVerbruggen** [comment]
@Menardi Thanks for figuring that out, I'm sure it will save one or two headaches for folks :+1: 

Added to [the doc](http://plugins.telerik.com/cordova/plugin/native-page-transitions) - look for 'crosswalk'

**Comment by bau720123** [comment]
hi all
try many times
for now，"CrosswalkAnimatable" seem not work in PGB
I have already post a question to ask PGB
http://community.phonegap.com/nitobi/topics/-crosswalkanimatable-not-already-support-by-pgb

**Comment by vagrant1991** [comment]
@Menardi  @EddyVerbruggen  

<preference name="CrosswalkAnimatable" value="true" /> 

it doesn't work for me,  on xwalkVersion 14+

**Comment by vagrant1991** [comment]
@bau720123  have you solved this problem?

**Comment by bau720123** [comment]
hi @vagrant1991 
for now PGB not already solve the problem (not support "CrosswalkAnimatable")
the only way for now is use cordova

**Comment by vagrant1991** [comment]
hi @EddyVerbruggen  i found you have changed the getBitmap() method, and add a method called findCrosswalkTextureView() , i updated the code. sometimes it works perfectly. but sometimes it doesn't. the imagview is black.  any idea?

**Comment by rdesimone** [comment]
The plugin is working nicely with Crosswalk on ARM devices but not on X86 device (Android 4.3).

Sometimes the slide transition executes, sometimes not - it is unpredictable. Played also with duration and delay to make sure the page is ready.

var options = {
            "href": target,
            "direction"        : direction, 
            "duration"         :  duration,
            "slowdownfactor"   :    -1, 
            "slidePixels"      :   20, 
            "iosdelay"         :  100, 
            "androiddelay"     :  delay,
            "winphonedelay"    :  250,
            "fixedPixelsTop"   :    0, 
            "fixedPixelsBottom":   0
        };

logcat does not display any message when the page does not slide. Also the plugin callbacks don't output anything, neither error or success. But when selecting a different tab, two transitions happens and logcat shows this:

03-25 16:45:02.707 29193-29193/com.application.myapp W/CordovaPlugin: Attempted to send a second callback for ID: NativePageTransitions957408125
                                                                            Result was: "OK"

Native Transition Plugin version: 0.6.3
Crosswalk Plugin: 1.6.1
Crosswalk 17.46.448.10
Cordova Android@5.1.1
Cordova CLI 6.0.0
Android 4.3 (ASUS)

Thanks for your help!

**Comment by rdesimone** [comment]
The problem was JS related. @EddyVerbruggen downgrading to kendoUI 2013.3.1503 (from 2016.1.322) solved the problem.

**Comment by ehorodyski** [comment]
I'm running into this issue now, using Ionic along with Native Page Transitions + Crosswalk on Android 6.0.

**Comment by duke111** [comment]
Ciao! 

Quando ho trovato questa  roba estremamente buona ho pensato che potrebbe piacervi, date  un'occhiata qui <http://daily.primocucina.com/oxw/8>

I miei migliori auguri, Fabio Fracassini

**Comment by AmitMY** [comment]
Also having this issue, using ionic 2
